### PR TITLE
feat: css variables

### LINKED ISSUE #159: feat: css variables
## Description

While the design package right now provides the solution for the CSS parts. In some simple use cases, like changing the theme color will be tiresome for the end user using the CSS pasts solution. It will be better for users to have a much easier solution for such small use cases.

## Possible solution(s)

 CSS variables can be used to solve this. There are many implementations using the CSS variables but the general idea will be to have a clear, concise & consistent variables across all components.

### PR DESCRIPTION
**Pull request by anuragxxd**
## Description

Fixes #159 

## Checklist

- [ ] My code follows the [contributing guidelines][contributing] of this project.
- [ ] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have updated the user-facing documentation to describe any new or changed behavior.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

## Summary by Sourcery

This pull request introduces a comprehensive set of CSS variables to standardize and simplify the styling across the project. It refactors existing styles to use these new variables and updates various components to ensure consistency. Additionally, it enhances the form, collection, and details components with new styles and updates the demo HTML files accordingly.

* **New Features**:
    - Introduced CSS variables for primary, success, warning, danger, and neutral colors, as well as various UI elements like border radius, shadows, spacing, transitions, fonts, and input styles.
* **Enhancements**:
    - Refactored existing styles to use new CSS variables for better maintainability and consistency.
    - Updated form, collection, and details components to utilize the new CSS variables.
    - Added new styles for form elements such as file inputs, switch labels, and submit buttons.
    - Enhanced the details component with new styles for actions, fields, and copy buttons.
* **Documentation**:
    - Updated demo HTML files to reflect the new CSS variable usage and removed inline styles.

### CODE REVIEW

**Comment by SalihuDickson** [comment]
Hey @anuragxxd, great work on this. Can you just add examples of how the css variables can be used to change the theme of the components in the demo so that new users can have something to play around with in plain HTML when they're trying to get a better understanding of how to style the components.

**Comment by anuragxxd** [comment]
> Hey @anuragxxd, great work on this. Can you just add examples of how the css variables can be used to change the theme of the components in the demo so that new users can have something to play around with in plain HTML when they're trying to get a better understanding of how to style the components.

Sure, There is still work left on the docs part. I will try to sort that out before the weekly meeting for better momentum.

**Comment by SalihuDickson** [comment]
hey @anuragxxd, you haven't provided a way to access the text in the editor. I think a simple way to handle this would be to have the component fire an event whenever the text is modified. If you want i can implement this myself incase you're too busy at the moment. I just didn't want to work on an implementation if you have already built this functionality but are yet to push it\.

**Comment by SalihuDickson** [comment]
hey @anuragxxd, could you please take a look at the above stated issue, it is blocker for a feature i am working on. Thank you!

**Comment by anuragxxd** [comment]
> hey @anuragxxd, could you please take a look at the above stated issue, it is blocker for a feature i am working on. Thank you!

Hey @SalihuDickson, was little busy but have pushed the changes now. Also, ace will also now show the errors too based on the language!

**Comment by anuragxxd** [comment]
Hi @SalihuDickson and @JaeAeich,
I believe I've completed all the work on this branch. Could you both please review it?
@SalihuDickson: Regarding the documentation, I have some changes on another branch. I'll update the docs in that PR, as this one has grown quite large.
@JaeAeich: Good news - this should unblock your code component integration PR.
Please let me know if you need any clarification or have any questions. Thanks!

**Review by JaeAeich — COMMENTED** [review]
LGTM!
Whole lib changed in this PR though, hopefully the design will be consistent from now on :crossed_fingers:, it was a necessary change. 
Please format the files though!

File: packages/ecc-utils-design/src/components/details/details.ts
Comment (SalihuDickson) [inline packages/ecc-utils-design/src/components/details/details.ts:212]: is there a reason why this needs inline style?

File: packages/ecc-utils-design/src/components/form/form.ts
Comment (SalihuDickson) [inline packages/ecc-utils-design/src/components/form/form.ts:1]: does the implementation of variables eliminate the need for parts? As I understand the parts allow for more targeted styling, while the variables provide more general styling for multiple components that use those variables. 

Can you please walk me through the thought process here, I might be missing something

**Review by SalihuDickson — CHANGES_REQUESTED** [review]

File: packages/ecc-utils-design/src/components/code/code.ts
Comment (anuragxxd) [reply packages/ecc-utils-design/src/components/code/code.ts:54]: Make sense! Will change

File: packages/ecc-utils-design/src/components/code/code.ts
Comment (anuragxxd) [reply packages/ecc-utils-design/src/components/code/code.ts:12]: Yeah I mean I thought about this, but this means that we provide support for these languages. User is free to use any language they want on their risks!

File: packages/ecc-utils-design/src/components/details/details.ts
Comment (anuragxxd) [reply packages/ecc-utils-design/src/components/details/details.ts:212]: Nope I just missed this! Thanks

File: packages/ecc-utils-design/src/components/form/form.ts
Comment (anuragxxd) [reply packages/ecc-utils-design/src/components/form/form.ts:1]: Hmm @SalihuDickson, good point. 

I don't think CSS parts are obsolete. Users might still need to target specific parts of a component for redesign, considering that a CSS variable might be used elsewhere too.

However, if our primary goal isn't precise customization, the need for CSS parts could be reduced. I believe a discussion about our priorities is necessary to decide this.

Comment (anuragxxd) [reply packages/ecc-utils-design/src/components/form/form.ts:1]: Let's merge this though!

Comment (SalihuDickson) [reply packages/ecc-utils-design/src/components/form/form.ts:1]: yeah, I agree that we should discuss this further, before deciding whether to extend or reduce the functionality. But I think we should maintain current functionality (revert the removal of the parts functionality) until then. Besides that this PR is just about ready to merge. 

Thank You.

File: packages/ecc-utils-design/demo/code/index.html
Comment (SalihuDickson) [inline packages/ecc-utils-design/demo/code/index.html:41]: this will need to be changed to `value`, since it was changed in the code component

**Review by SalihuDickson — CHANGES_REQUESTED** [review]

File: packages/ecc-utils-design/src/components/form/form.ts
Comment (anuragxxd) [reply packages/ecc-utils-design/src/components/form/form.ts:1]: Yeah actually upon further consideration, I've decided to remove the CSS parts functionality. Here's my reasoning:

- Consistency: The usage of parts is inconsistent across our codebase, making maintenance challenging.
- Export pipeline issues: The current process for exporting parts to other packages is suboptimal and needs improvement.

However, I understand this is a significant change. My goal is to streamline our CSS approach while ensuring we meet all necessary styling requirements. Moreover, if we decide on keeping the parts in the future the contributor working on it will have the reference in VCS history. Hope that sums up why I have removed parts here. It is too much of a headache to even see in our codebase.

File: packages/ecc-utils-design/demo/code/index.html
Comment (anuragxxd) [reply packages/ecc-utils-design/demo/code/index.html:41]: Thanks for the catch

File: packages/ecc-utils-design/src/components/form/form.ts
Comment (SalihuDickson) [reply packages/ecc-utils-design/src/components/form/form.ts:1]: understandable, but you do understand that without any sort of CSS parts implementation we would be severely restricting the stlying options we provide for each component. @JaeAeich, any thoughts on this?

Comment (anuragxxd) [reply packages/ecc-utils-design/src/components/form/form.ts:1]: Yes, I understand that. The reason I removed the CSS implementation is that it wasn't up to standard and wasn't usable in its current form. This doesn't mean we won't include it; it simply means that we can add it from scratch after this PR is merged. It just isn't within the scope of this PR.

**Review by SalihuDickson — APPROVED** [review]

**Pull request merged**
